### PR TITLE
TASK: Include AFX in development collection

### PR DESCRIPTION
**Pull request by kitsunet**
This merges the AFX repository as subtree of the development collection, additionally the splitter configuration will have to be adapted afterwards and the Neos.Fusion.AFX repository made read-only.

### CODE REVIEW

**Review by jonnitto — APPROVED** [review]
Fine (by reading)

**Review by mficzel — COMMENTED** [review]
I think the afx tests still have to be included into the neos testsuite. Cannot find code for that.

**Comment by kitsunet** [comment]
> I think the afx tests still have to be included into the neos testsuite. Cannot find code for that.
@mficzel not ssure what you mean?

**Comment by mficzel** [comment]
@kitsunet the afx tests run by travis should be included into the unit and functional testsuite of the dev-collection and the travis.yaml from the package removed.

Not totally sure how to proceed with https://github.com/neos/fusion-afx/pull/35. Would be good to have this in here aswell but it can always be done later.

**Comment by kitsunet** [comment]
ohhhh, right, that makes sense, taking care.

**Comment by kdambekalns** [comment]
>  Neos.Fusion.AFX repository made read-only.

Or rather, become the target of the split job, no?

**Comment by kitsunet** [comment]
> > Neos.Fusion.AFX repository made read-only.
> 
> Or rather, become the target of the split job, no?

yes of course :)

**Comment by kdambekalns** [comment]
BTW: I like the branch name! 🤣

File: Neos.Fusion.Afx/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/vendor
```
Comment (kdambekalns) [inline Neos.Fusion.Afx/.gitignore:1]: Is this file needed at all?

File: Neos.Fusion.Afx/.styleci.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+preset: psr2
```
Comment (kdambekalns) [inline Neos.Fusion.Afx/.styleci.yml:1]: Is this file needed at all?

Comment (kitsunet) [reply Neos.Fusion.Afx/.styleci.yml:1]: only in case you want to run styleci separately on the split repo I guess

Comment (kitsunet) [reply Neos.Fusion.Afx/.styleci.yml]: removed

File: Neos.Fusion.Afx/.gitignore (hunk 1 above)
Comment (kitsunet) [reply Neos.Fusion.Afx/.gitignore]: removed

File: Neos.Fusion.Afx/composer.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "neos/fusion-afx",
+  "description": "JSX inspired compact syntax for Neos.Fusion",
+  "type": "neos-package",
+  "license": "GPL-3.0-or-later",
+  "replace": {
+    "packagefactory/atomicfusion-afx": "*"
+  },
+  "require": {
+    "neos/fusion": "^4.2 || ^5.0 || dev-master"
```
Comment (albe) [inline Neos.Fusion.Afx/composer.json:10]: Needs to be `|| ^6.0` :slot_machine:😉 too then. Or we go with `"*"` or `"self.version"` as in some of the other packages.

Comment (kdambekalns) [reply Neos.Fusion.Afx/composer.json:10]: Good point!

**Comment by markusguenther** [comment]
Retrigger travis

File: composer.json [hunk 4]
```diff
@@ -72,6 +71,9 @@
             "Neos\\Diff\\": [
                 "Neos.Diff/Classes"
             ],
+            "Neos\\Fusion\\Afx\\": [
+                "Neos.Fusion.Afx/Classes/"
```
Comment (mficzel) [inline composer.json:75]: Probably irrelevant but all other packages do use the classes path without trailing slash?

File: Neos.Fusion.Afx/composer.json (hunk 3 above)
Comment (mficzel) [reply Neos.Fusion.Afx/composer.json:10]: I agree that this should be '*' or 'self.version'

**Comment by mficzel** [comment]
I verified that it works and looks good and the tests are running on the dev-collection. So other that the composer cross-dependency question this is fine and should go in.

File: composer.json (hunk 4 above)
Comment (kitsunet) [reply composer.json:75]: Should not matter but I will adapt it

File: Neos.Fusion.Afx/composer.json (hunk 3 above)
Comment (kitsunet) [reply Neos.Fusion.Afx/composer.json:10]: Yep, right

**Review by mficzel — APPROVED** [review]
